readthedocs/core/utils/__init__.py

```diff
@@ -102,6 +104,7 @@ def prepare_build(
     kwargs = {
         'record': record,
         'force': force,
+        'commit': commit,
```

Comment:
I think we should also be passing `commit` into the Build object here when created, if we have it.

Reply:
Done!